### PR TITLE
perf: the dynamic Dory commitment computation should have a GPU implementation

### PR DESCRIPTION
**Pull request by jacobtrombetta**
# Rationale for this change
The Blitzar project implemented a `vlen_msm` function that is optimized for the dynamic Dory structure. This PR creates the `dynamic_dory_commitment_helper_gpu` module when using the blitzar feature flag. Helper functions are added that create the input needed for Blitzar's vlen_msm function.

This work is the final piece of a PR that was split up: https://github.com/spaceandtimelabs/sxt-proof-of-sql/pull/231.

# What changes are included in this PR?
- The GPU implementation of the dynamic Dory commitment computation is implemented.
- `#[allow(dead_code)]` tags are removed from the `dynamic_dory_structure` module.
- `pack_scalars::min_as_f` is made public - this will likely need refactored in a follow up PR when the `compute_dynamic_T_vec_prime` is ported to the GPU.

# Are these changes tested?
Yes

### CODE REVIEW

File: crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+    // Create scalars array. Note, scalars need to be stored in a column-major order.
+    let num_scalar_rows = max_width;
+    let num_scalar_columns =
+        (num_of_bytes_in_committable_columns + ones_columns_lengths.len()) * max_height;
+    let mut blitzar_scalars = vec![0u8; num_scalar_rows * num_scalar_columns];
+
+    // Populate the scalars array.
+    let span = span!(Level::INFO, "pack_blitzar_scalars").entered();
+    if !blitzar_scalars.is_empty() {
```
Comment (JayWhite2357) [inline crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs:183]: NIT: this `if` doesn't do anything. The chunks_exact_mut will be an empty iterator if it is empty.

**Review by JayWhite2357 — APPROVED** [review]
Minor NIT. Otherwise looks good.

**Review by JayWhite2357 — DISMISSED** [review]

File: crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs (hunk 1 above)
Comment (JayWhite2357) [reply crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs:183]: Never mind, this doesn't work.

**Review by JayWhite2357 — DISMISSED** [review]

**Review by tlovell-sxt — APPROVED** [review]
Reviewing the changes since last approval - it looks like it's just handling empty tables more explicitly?

Took me a sec to understand the already-approved code but LGTM

**Pull request merged**